examples/react/basic/.eslintrc

```diff
@@ -1,7 +1,8 @@
 {
   "extends": ["react-app", "prettier"],
+  "plugins": ["@tanstack/eslint-plugin-query"],
   "rules": {
-    // "eqeqeq": 0,
-    // "jsx-a11y/anchor-is-valid": 0
+    "@tanstack/query/exhaustive-deps": "error",
+    "@tanstack/query/prefer-query-object-syntax": "error"
   }
```

Comment:
is there a "recommended" ruleset? how would I use it?

Reply:
would this be:

```
"extends": ["@tanstack/eslint-plugin-query/recommended"]
```

?

Just trying to get this right for the docs 😅

Reply:
Yes:

```
{
  "extends": ["plugin:@tanstack/eslint-plugin-query/recommended"]
}
```

Reply:
are you sure it's `@tanstack/eslint-plugin-query` and not `@tanstack/query` ? I'm always confused by when to include `eslint-plugin` and when not 😅

Reply:
the rules in the example say:

```
"@tanstack/query/exhaustive-deps": "error",
```

Reply:
also: does it need the `typescript-eslint` parser to be present, or will it add it manually? Do the rules only work on TypeScript files?

Sorry for all the questions, just trying to write something for the docs and want to understand it properly 😅

Reply:
I understand why it doesn't look very clear, It depends on the plugin's package name. Whether it's a scoped package or a standard name (in our case it's scoped).

the pattern of `extends` is "plugin:{package_name}/{config_name}"
the pattern of `rules` is: "{package_name minus prefix 'eslint-plugin-'}/{rule_name}

`eslint-plugin-` is omitted since it's redundant.

You don't have to pass `typescript-eslint` as a parser, and it won't be added manually. The plugin now works with babel's parser (which is the default parser).

Reply:
Thanks for explaining ❤️ 

I've added the docs and moved the rule specific docs to our docs structure as well. I think it's good to go ?

Reply:
Let's 🚀 it